### DISCUSSION TITLE
post/input-driven-resampling-with-naudio-using-acm

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#45] [Announcements]
# post/input-driven-resampling-with-naudio-using-acm

Mark Heath's Development Blog

https://markheath.net/post/input-driven-resampling-with-naudio-using-acm

**Comment by zmrcic** [comment]
Hi...great library. At the moment I'm saving rtp to wav file, but can not manage to use acmstream to forward rtp to speakers...
how to detect rate, bits from incoming rtp packege?
thx

Dim resampleStream As New NAudio.Wave.Compression.AcmStream(New WaveFormat(10000, 8, 1), New WaveFormat(8000, 16, 1))

**Reply by markheath** [reply]
Afraid I don't know how sample formats are reported over RTP, but I would guess that 8kHz mono mu-law or a-law might be a likely encoding for voice communications
